packages/hub/src/scope.ts

```diff
@@ -63,11 +63,16 @@ export class Scope {
    */
   protected async notifyEventProcessors(
     event: SentryEvent,
-  ): Promise<SentryEvent> {
-    return this.eventProcessors.reduce(async (prev, callback) => {
-      const prevEvent = await prev;
-      return callback(prevEvent);
-    }, Promise.resolve(event));
+  ): Promise<SentryEvent | null> {
+    let processedEvent: SentryEvent | null = event;
+    for (const processor of this.eventProcessors) {
```

Comment:
I'd wrap every `await processor(processedEvent)` call in `try/catch`, as I think we should let it run if one of processors fail anyway

```js
for (const processor of this.eventProcessors) {
  try {
    processedEvent = await processor({ ...processedEvent });
    if (processedEvent === null) {
      // tslint:disable-next-line:no-null-keyword
      return null;
    }
  } catch (e) {
    continue;
  }
}
```

(`processor({ ...processedEvent })` is not necessary, but may be useful, to make sure we never mutate, as processors that mutate AND throw, may leave event in a strange state)